app/models/task.py

```diff
@@ -1,5 +1,38 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import String, DateTime, ForeignKey
+from datetime import datetime
+from typing import Optional
 from ..db import db
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] = mapped_column(String(50))
+    description: Mapped[str] = mapped_column(String(255))
+    completed_at: Mapped[Optional[datetime]] = mapped_column(DateTime)
+    goal_id: Mapped[Optional[int]] = mapped_column(ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = relationship(back_populates="tasks")
+
+    def to_dict(self):
+        task_dict = {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": bool(self.completed_at)
+        }
+
+        if self.goal_id is not None:
+            task_dict["goal_id"] = self.goal_id
+
+        return task_dict
+
+    @classmethod
+    def from_dict(cls, data):
+        data["completed_at"] = data.get("completed_at", None)
+
+        task = Task(
+            title=data["title"],
+            description=data["description"],
+            completed_at=data["completed_at"],
+        )
+
+        return task
```

Comment:
Since you don't reference the variable `task` after you initialize it and before you return it, you can just directly return the instance of task without making a variable:
```suggestion
        return cls( 
            title=data["title"],
            description=data["description"],
            completed_at=data["completed_at"],
        )
```